Tests/Unit/HttplugFactoryTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Http\HttplugBundle\Tests\Unit\ClientFactory;
+
+use Http\HttplugBundle\Collector\MessageJournal;
+use Http\HttplugBundle\HttplugFactory;
+use Puli\Discovery\Api\Discovery;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttplugFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testEvaluateCondition()
+    {
+        $method = new \ReflectionMethod(HttplugFactory::class, 'evaluateCondition');
```

Comment:
Yes. You can mock the hasParameter and getParameter methods of the class binding.

Generally you should always test the public API. It is part of the public API that if a class binding has a depends parameter, it gets evaluated "somehow".

So what you really want to test in this case is what separate depends parameters return a valid class binding, which of them end up in an exception.